### PR TITLE
feat (examples): add streamObject example with streamableValue

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary
Adds a reference example of how to use `streamObject` and `streamableValue` together for generative UI.

## Demo
CleanShot 2024-06-01 at 11 09 17

### CODE REVIEW

File: examples/next-openai/app/stream-object/page.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+'use client';
+
+import { StreamableValue, useStreamableValue } from 'ai/rsc';
+import { useState } from 'react';
+import { generateNotifications } from './actions';
+import { PartialNotification } from './schema';
+
```
Comment (nicoalbanese) [inline examples/next-openai/app/stream-object/page.tsx:7]: add this:
// Force the page to be dynamic and allow streaming responses up to 30 seconds
export const dynamic = 'force-dynamic';
export const maxDuration = 30;

**Review by nicoalbanese — APPROVED** [review]

File: examples/next-openai/app/stream-object/page.tsx (hunk 1 above)
Comment (lgrammel) [reply examples/next-openai/app/stream-object/page.tsx:7]: thanks yes

File: examples/next-openai/app/stream-object/actions.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'use server';
```
Comment (MaxLeiter) [inline examples/next-openai/app/stream-object/actions.ts:1]:
```suggestion

```
This is unnecessary if there's `use server` in the function.

**Review by MaxLeiter — CHANGES_REQUESTED** [review]

File: examples/next-openai/app/stream-object/actions.ts (hunk 2 above)
Comment (lgrammel) [reply examples/next-openai/app/stream-object/actions.ts:1]: thank you!

**Comment by lgrammel** [comment]
@MaxLeiter removing `use server` broke the example:

CleanShot 2024-06-02 at 13 27 59@2x

I'll put it back in.

File: examples/next-openai/app/stream-object/actions.ts (hunk 2 above)
Comment (lgrammel) [reply examples/next-openai/app/stream-object/actions.ts:1]: put it back in, see comment about broken example

**Comment by MaxLeiter** [comment]
@lgrammel then you can remove the one inside the function. If that doesn’t work we should report upstream to Next

File: examples/next-openai/app/stream-object/actions.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import { openai } from '@ai-sdk/openai';
+import { streamObject } from 'ai';
+import { createStreamableValue } from 'ai/rsc';
+import { PartialNotification, notificationSchema } from './schema';
+
+export async function generateNotifications(context: string) {
+  'use server';
```
Comment (MaxLeiter) [inline examples/next-openai/app/stream-object/actions.ts:9]:
```suggestion
```

**Review by MaxLeiter — APPROVED** [review]

**Comment by lgrammel** [comment]
@MaxLeiter thanks - that worked

**Pull request merged**